### PR TITLE
Add standard metadata build labels 

### PR DESCRIPTION
**Pull request by jason-fox**
- Run as anonymous user
- Add standard metadata build `LABEL` (see [opencontainers.org](https://github.com/opencontainers/opencontainers.org))
- Remove **deprecated** `MAINTAINER` - replace with `LABEL` 

Similar to, but less comprehensive than https://github.com/telefonicaid/iotagent-ul/pull/435

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -128,3 +152,19 @@ WORKDIR /
 ENTRYPOINT ["/usr/bin/contextBroker","-fg", "-multiservice", "-ngsiv1Autocast" ]
 EXPOSE 1026
 
+LABEL "maintainer"="FIWARE IoTAgent Team. Telefónica I+D"
```
Comment (fgalan) [inline docker/Dockerfile:155]:
```suggestion
LABEL "maintainer"="Orion Team. Telefónica I+D"
```

File: docker/hooks/build [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
```
Comment (fgalan) [inline docker/hooks/build:1]: Who is using this hook? dockerhub?

Comment (jason-fox) [reply docker/hooks/build:1]: Assuming you are building this in the same manner as the IoT Agents in **Docker Cloud.**  the`SOURCE_BRANCH` **Env** variable will be available. The `build` file will provide additional `org.opencontainers` labels. 

See: https://github.com/telefonicaid/iotagent-ul/issues/317#issuecomment-446630979

Comment (fgalan) [reply docker/hooks/build:1]: NTC

**Comment by fgalan** [comment]
We are doing some tests with the new Dockefile. Once finished we will provide feedback or eventually merge the PR.

CC: @IvanHdzC

**Review by IvanHdzC — APPROVED** [review]
LGTM

File: docker/hooks/build [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+#
+
+docker image build --build-arg SOURCE_BRANCH=$SOURCE_BRANCH \
+	--label "org.opencontainers.image.created"=$(date +%Y-%m-%d) \
+	--label "org.opencontainers.image.revision"=$SOURCE_COMMIT \
+	--label "org.opencontainers.image.version"=$DOCKER_TAG \
+	-t $IMAGE_NAME \
```
Comment (fgalan) [inline docker/hooks/build:27]: Based in a recent experience solving a build problem in release/2.4.0 branch (see PR https://github.com/telefonicaid/fiware-orion/pull/3702), we suggest the following:

```suggestion
# -e, to end execution immediately if 'cd ..' fails
# -x, to print output as we go
set -ex

# due to Dockerfile is not in the repository root, but in a subdirectory and we use -f docker/Dockerfile
cd ..

docker image build --build-arg GIT_REV_ORION=$SOURCE_BRANCH \
	--label "org.opencontainers.image.created"=$(date +%Y-%m-%d) \
	--label "org.opencontainers.image.revision"=$SOURCE_COMMIT \
	--label "org.opencontainers.image.version"=$DOCKER_TAG \
	-f docker/Dockerfile \
	-t $IMAGE_NAME \
```

CC: @IvanHdzC @AlvaroVega

Comment (fgalan) [reply docker/hooks/build:27]: @jason-fox if you prefer, we can do it the other way around: backport PR https://github.com/telefonicaid/fiware-orion/pull/3702 to master, then adapt your PR to the new file in master.

What do you think?

**Review by fgalan — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**